cpp/ql/src/Security/CWE/CWE-704/IncorrectTypeConversion.ql

```diff
@@ -0,0 +1,30 @@
+/**
+ * @name Cast between semantically different integer types: HRESULT to/from a Boolean type
+ * @description Cast between semantically different integer types: HRESULT to/from a Boolean type.
+ *              Boolean types indicate success by a non-zero value, whereas success (S_OK) in HRESULT is indicated by a value of 0. 
+ *              Casting an HRESULT to/from a Boolean type and then using it in a test expression will yield an incorrect result.
+ * @kind problem
+ * @id cpp/incorrect-type-conversion
```

Comment:
I think this ID and the file name of the query are too generic. It's a very specific query about conversions between HRESULT and bool, and the names should reflect that.

Reply:
Sounds good. How about "cpp/hresult-to-boolean-conversion" (please feel free to suggest another one if you prefer).